### ISSUE TITLE
Import statements taking time on spark (Pyspark)

### ISSUE THREAD

**Issue by anishsharma** [keras-team/keras#11338]
I am developing a python prediction script using spark(Pyspark) streaming and keras. The prediction is happening on the executor where I am calling model.predict(). 

Modules that I have imported are

`from keras.layers.core import Dense, Activation, Dropout
from keras.layers.recurrent import LSTM
from keras.models import Sequential`

I have checked and these imports are taking 2.5 seconds on spark driver(2 core+ 2gb) to load. What is surprising for me is that each time executor gets the job, it automatically do these imports again. The raeason I am sure that these imports are happening each time a job is submitted to the executor is because I see below statements in executor logs per job which are only comes when I do imports of above said modules.

`/opt/conda/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.`

My target is to make the prediction in 1 sec but imports itself are taking 2.5 seconds (Each time imports happens on the spark executor). Is this is the intended behavior? Is their anything I can do to minimize this ti say in milliseconds ?

**Comment by maxpumperla** [comment]
@anishsharma are you sure it's the imports that take that long? how do you measure that? Could be loading the tensorflow backend.

Re your question on imports, of course the executors need to import your dependencies for each job (and execute whatever else you give them). Think of it as working on a fresh shell locally. you install libraries once, but you need to import them each time you use them. Spark serializes your Python and deserializes it on executors (using py4j) after all.

For more details I'd have to know your setup, but loading the models once (slow) followed by a few batch predictions is definitely possible.

**Comment by gabrieldemarmiesse** [comment]
Closing this issue as the long import is very likely `import tensorflow`. There is nothing we can do about that. You can transfer this issue to the tensorflow repository if necessary.

**Comment by anishsharma** [comment]
> @anishsharma are you sure it's the imports that take that long? how do you measure that? Could be loading the tensorflow backend.
> 
> Re your question on imports, of course the executors need to import your dependencies for each job (and execute whatever else you give them). Think of it as working on a fresh shell locally. you install libraries once, but you need to import them each time you use them. Spark serializes your Python and deserializes it on executors (using py4j) after all.
> 
> For more details I'd have to know your setup, but loading the models once (slow) followed by a few batch predictions is definitely possible.

@maxpumperla I just put time.time() before and after imports and checked the difference. It was 2.5 secs but it does says "Using Tensorflow backend" at console. I guess it is tensorflow which is taking time to load and not keras.
